### DISCUSSION TITLE
Unable to create foreign key on user auth table

### DISCUSSION THREAD

**Discussion by barbalex** [supabase/supabase#11827] [Questions]
# Bug report

## Describe the bug

In my db there is this table:
```sql
CREATE TABLE users (
  id uuid PRIMARY KEY DEFAULT gen_random_uuid (),
  email text UNIQUE DEFAULT NULL,
  auth_user_id uuid DEFAULT NULL,
  client_rev_at timestamp with time zone DEFAULT now(),
  client_rev_by text DEFAULT NULL,
  server_rev_at timestamp with time zone DEFAULT now(),
  deleted boolean DEFAULT FALSE
);
```

I tried to add a foreign key to the auth.users table using this:
```sql
ALTER TABLE "public"."users"
  ADD FOREIGN KEY ("auth_user_id") REFERENCES "auth"."users" ("id");
```

It worked fine in the local studio url (http://localhost:54323/project/default/sql). It fails on the remote server in two different ways:

- in the remote studio (https://app.supabase.io/project/xescznrwxlrclwapyuzm/sql) with this error: `relation "auth.users" does not exist`
- when run in pgAdmin (connected as user postgres): `ERROR:  must be owner of table users`

## System information

- OS: windows 11
- Browser: chrome up to date
- Version of supabase-js: 1.33.2
- Version of Node.js: 17.3.1

**Comment by joshenlim** [comment]
Hey there - I've just tried to reproduce on production but can't seem to
If this is still happening for you could you open a support ticket [here](https://app.supabase.io/support/new)? It may be something specifically for your project

I'll go ahead to close this issue for now 🙏

**Comment by barbalex** [comment]
This is still happening persistently. Combined with the fact that local development does not work due to consistent problems with migrations failing and the server running out of disk space, it seems that supabase is not an option for this project. Which is a REAL pitty as it seemed so great.

**Comment by barbalex** [comment]
I destroyed the first project and rebuilt with a pro plan. Exact same issue.

**Comment by kiwicopple** [comment]
> This is still happening persistently

Hey @barbalex - I tried to replicate your issue again and I didn't get an error:

https://user-images.githubusercontent.com/10214025/175794227-e370aeea-c86a-4c91-86d8-bf3156734a5f.mov

Are you running anything that might be changing the permissions of your schemas?

> rebuilt with a pro plan

How are you "rebuilding"? Are you using a migration tool?

> and the server running out of disk space

How much data are you loading into your instance? The disk will grow - we have customers with terrabytes of data operating fine. It might be that you're loading up data faster than the disk can grow

**Comment by hf** [comment]
Make sure that the user/role executing the `ALTER TABLE` statements has at least `SELECT` permissions on the `auth` and `public` schemas.

Use `SELECT current_user;` or `SELECT current_role;` to see them. On a standard Supabase project these typically are `supabase_admin`.

Since this is an issue that we're having trouble reproducing I'll convert this to a discussion.
